Automattic/newspack-plugin
# Modal checkout: allow covering Stripe's fees

### All Submissions:






### Changes proposed in this Pull Request:



Adds the option for the donor to cover Stripe's transaction fees:

<img width="558" alt="image" src="https://github.com/Automattic/newspack-plugin/assets/7383192/1f7b49e4-4912-45b1-bf8e-0da3bce0ba9c">


See 1200133283036252-as-1205668937699532

The feature is on by default. To opt-out, a change the "Allow donors to cover transaction fees" setting in Reader Revenue wizard, Stripe Gateway tab.

There's one non-covered edge case – use of coupons. The logic to handle covering fees with coupons applied would make this code overly complex, and since we're talking about donations here, usage of coupons seems like an extreme edge case. If coupons are applied to the order, the feature will disappear gracefully. 

### How to test the changes in this Pull Request:

1. Make sure you have "Newspack" set as the Reader Revenue platform and `Stripe (Credit Card)` enabled as a payment gateway
2. Visit a page with a donation prompt, and initiate the flow to trigger the modal checkout
3. Observe a "Cover transaction fees?" checkbox above the card details inputs in the "Payment info" section of the checkout
4. Complete transactions with and without opting to cover the fees, observe the WC orders in the latter case:
    - have the total order amount increased to offset the Stripe fee
    - have a `newspack_donor_covers_fees` meta field with value of `1`
    - have an order note attached commenting about the fact of covering fees
5. In WC settings, turn on coupons and create a coupon. Use the coupon in the checkout – observe that the feature is not active when the coupon is applied
6. Visit Newspack's Reader Revenue wizard, Stripe Gateway tab. Update the fee calculation inputs and observe these are reflected on the front-end.

### Other information:
